pkg/proxy/proxy.go

```diff
@@ -228,21 +284,27 @@ func fillIdentity(info *accesslog.EndpointInfo, id policy.NumericIdentity) {
 // endpoint info based on ip address, as the ip address would be that of the
 // egress proxy i.e host.
 func fillIngressSourceInfo(info *accesslog.EndpointInfo, ip *net.IP, srcIdentity uint32) {
-	if ip != nil {
-		if ip.To4() != nil {
-			info.IPv4 = ip.String()
-		} else {
-			info.IPv6 = ip.String()
-		}
-	}
 
 	if srcIdentity != 0 {
+		if ip != nil {
+			if ip.To4() != nil {
+				info.IPv4 = ip.String()
+			} else {
+				info.IPv6 = ip.String()
+			}
+		}
 		fillIdentity(info, policy.NumericIdentity(srcIdentity))
 	} else {
-		// source security identity 0 is possible when somebody else other than the BPF datapath attempts to
+		// source security identity 0 is possible when somebody else other than
+		// the BPF datapath attempts to
 		// connect to the proxy.
-		// We should log no source information in that case, in the proxy log.
-		log.Warn("Missing security identity in source endpoint info")
+		// We should try to resolve if the identity belongs to reserved_host
+		// or reserved_world.
+		if ip != nil {
+			fillEndpointInfo(info, *ip)
+		} else {
+			log.Warn("Missing security identity in source endpoint info")
+		}
```

Comment:
I might not understand the code but doesn't fillEndpointInfo do all this logic now? It correctly fills out ipv4 or ipv6 IPs, and it will populate the NumericIdentity (although it casts it into uint64 and not `policy.NumericIdentity`).

Reply:
This case/warning is when we do not have the src Identity field, as well as the IP.

Reply:
Also if you are referring to `info.Identity = uint64(ep.GetIdentity())` in `fillEndpointInfo`, the type of info.Identity is uint64 and hence the typecast.

Reply:
I had meant the entire srcIdentity != 0 if else clause but reading it again that's clearly wrong. I had thought that fillEndpointInfo would handle the srcIdentity == 0 situation but it doesn't. Anyway, ignore me!